frontend/src/components/blog-details/blog.details.jsx

```diff
@@ -1,39 +1,31 @@
 import { BlogSections, AboutAuthor, PageHeader } from "shared-components";
 import BlogsService from "pages/api/blogsService";
 import { useState, useEffect } from "react";
-import { useRouter } from "next/router";
 
 export const BlogDetails = ({ blogId }) => {
-	const [blog, setBlog] = useState(null);
+  const [isLoadingBlog, setIsLoadingBlog] = useState(true);
+  const [blog, setBlog] = useState(null);
 
-	const router = useRouter();
+  async function getBlog() {
+    await BlogsService.getBlogById(blogId).then((response) => {
+      setBlog(response.data);
+      setIsLoadingBlog(false);
+    });
+  }
 
-	useEffect(() => {
-		async function getBlog(blogId) {
-			try {
-				const response = await BlogsService.getBlogById(blogId);
-				setBlog(response.data);
-			} catch (error) {
-				if (error.response.status === 404) await router.replace("/404");

```

Comment:
Sta se desi ako se pokusa pristupiti blogu sa ID-jem koji ne postoji?
Nisam siguran sta se desi ako WEB API vrati error response (s obzirom na to da je definisan `api.interceptors.response.use(...` u `src/pages/api/baseApi.js`).
Uglavnom, svakako bi trebalo da se prikaze stranica 404 ako blog sa tim ID-jem ne postoji.

Reply:
Prebacio sam sve u interceptors, mislim da je najbolje da tako handlujemo, da ne pisemo jedan isti kod na vise mesta, pa cemo kasnije stilizovati 401 i 404 stranice